### PR TITLE
:construction_worker:  Added noxfile for format & CI

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description

Added tesing in setup.cfg
Added a noxfile with fomat, lint & mypy session
Added CI for lint & mypy
Changed fromat.yml to use nox

## Checklist

- [ ] - All code changes have been tested, if any
- [ ] - The documentation has been updated accordingly for any code changes
- [ ] - This PR fixes an already existing issue
- [ ] - This PR adds a new feature (new functions, methods, classes, parameters, etc.)
- [ ] - This PR fixes a bug
- [ ] - This PR is a breaking change (removes/renames functions, parameters etc.)
- [x] - This PR is NOT a code change (changes Documentation files, README.md, CONTRIBUTING.md etc.)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -31,3 +31,5 @@ python_requires = >=3.8
 voice =
     opuslib
     pynacl
+testing =
```
Comment (TheUntraceable) [inline setup.cfg:34]: Maybe a workflows name?

**Review by TheUntraceable — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (Sigmanificient) [reply setup.cfg:34]: i mean it's gotta be  used for testing in general sooo

Comment (TheUntraceable) [reply setup.cfg:34]: I mean in my opinion it looks like it's just a dummy name we made up for a quick test. Maybe formatters?

Comment (Sigmanificient) [reply setup.cfg:34]: > ust a dummy name we

Well testing will regroup the nox suite & pytest, for workflows and user-side testing
Cant really think of a better name

Comment (TheUntraceable) [reply setup.cfg:34]: Formatting, Workflows, Checks, Actions

File: setup.cfg [hunk 2]
```diff
@@ -1,16 +1,16 @@
 [metadata]
 name = EpikCord.py
-version = attr: EpikCord.__version__
+version = attr: EpikCord.__init__.__version__
```
Comment (TheUntraceable) [inline setup.cfg:3]: Is this needed? Surely because __version__ is in the __init__.py, it is exported by default and this isn't needed?

**Review by TheUntraceable — APPROVED** [review]

File: setup.cfg (hunk 2 above)
Comment (Sigmanificient) [reply setup.cfg:3]: No it's not just a test x)
Let me revert this

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from setuptools import setup
+
+if __name__ == '__main__':
+    setup()
```
Comment (TheUntraceable) [inline setup.py:4]: Complete?

Comment (Sigmanificient) [reply setup.py:4]: Yup it will loke automatically in the setup.cfg file

**Pull request merged**